Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -138,9 +138,173 @@ public class PushDeploymentController : ApiController
+                    DoFullBuildByDefault = false,
+                    Message = message,
+                    WatchedFileEnabled = false 
+                };
+
+                switch (type)
+                {
+                    case Constants.War:
+                        if(website_framework!=Constants.Tomcat)
+                        {
+                           
+                            response.StatusCode = HttpStatusCode.BadRequest;
+                            response.Content = new StringContent("WAR files cannot be deployed to " + website_framework + " web apps.");
+                            
+                            return response;
+                        }
+
+                        //When the file name is constant, such as app.war or app.jar, deploymentInfo.FileName can be hardcoded
+                        //If the target directory for the file is wwwroot, deploymentInfo.TargetPath does not need to be set 
+                        deploymentInfo.FileName = "app.war";
+
+                        break;
+
+                    case Constants.Jar:
+                        if (website_framework != Constants.JavaSE)
+                        {
+
+                            response.StatusCode = HttpStatusCode.BadRequest;
+                            response.Content = new StringContent("JAR files cannot be deployed to " + website_framework + " web apps.");
+                            
+                            return response;
+                        }
+                        deploymentInfo.FileName = "app.jar";
+
+                        break;
+
+                    case Constants.Lib:
+                        if (string.IsNullOrWhiteSpace(path))
+                        {
+                            response.StatusCode = HttpStatusCode.BadRequest;
+                            response.Content = new StringContent("Path must be defined for library file deployments");
+                            
+                            return response;
+                        }
+                        else
+                        {
+
+                            //When the file name is unknown, then deploymentInfo.FileName must be derrived from the path query string
+                            deploymentInfo.FileName = Path.GetFileName(path);
+
+                            //If a path is provided, then deploymentInfo.TargetPath must also be derrived from the path query string
+                            deploymentInfo.TargetPath = Path.GetDirectoryName(path);
+                        }
+
+                        break;
+
+                    case Constants.Startup:
+
+                        //assumes windows application for Kudu deploymnets
+                        //deploymentInfo.TargetPath is set here because the files are not getting deployed to wwwwroot
+                        deploymentInfo.TargetPath = _environment.RootPath;
+                        deploymentInfo.FileName = "startup.bat";
```

Comment:
This can be startup.bat or startup.sh really. Better to have a helper function: GetStartupFileName() which can return startup.bat or startup.sh based on the platform. To detect the platform use OSDetector.IsOnWindows()